File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/reading_data/sample12.py
import dv_processing as dv

# Open a file
reader = dv.io.MonoCameraRecording("path/to/file.aedat4")

# Run the loop while stream contains data
while reader.isRunning():
    # Read a batch of IMU data from the camera
    imu_batch = reader.getNextImuBatch()
    if imu_batch is not None and len(imu_batch) > 0:
        # Print the info of the imu data
        print(f"Received {len(imu_batch)} IMU measurements")

File: source_code/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/reading_data/sample12.cpp
#include <dv-processing/io/mono_camera_recording.hpp>

int main() {
	// Open a file
	dv::io::MonoCameraRecording reader("path/to/file.aedat4");

	// Run the loop while data is available
	while (reader.isRunning()) {
		// Read IMU measurement batch, check whether it is correct.
		if (const auto imuBatch = reader.getNextImuBatch(); imuBatch.has_value() && !imuBatch->empty()) {
			// Print IMU batch information
			std::cout << "Received " << imuBatch->size() << " IMU measurements" << std::endl;
		}
	}

	return 0;
}
